src/compiler/checker.ts

```diff
@@ -10664,6 +10729,12 @@ namespace ts {
             }
 
             forEach(node.members, checkSourceElement);
+
+            // Classes containing abstract methods must be marked abstract
+            if (!(node.flags & NodeFlags.Abstract) && forEach(node.members, element => element.flags & NodeFlags.Abstract)) {
+                error(node, Diagnostics.Classes_containing_abstract_functions_must_be_marked_abstract);

```

Comment:
'containing abstract functions' should be 'containing abstract members' (or '... methods or constructors')